library/Erfurt/Store/Adapter/Stardog.php

```diff
@@ -0,0 +1,726 @@
+            throw new Erfurt_Store_Adapter_Exception('No graph URI given.');
+        }
+        if ($distinct) {
+            $distinct = "DISTINCT";
+        } else {
+            $distinct = "";
+        }
+
+        // more error save
+        if ($countSpec == '?*') {
+            $countSpec = '*';
+        }
+
+        $fromSpec = implode('> FROM <', (array)$graphUris);
+        $countQuery = sprintf(
+            'SELECT COUNT %s (%s) FROM <%s> %s',
+            $distinct,
+            $countSpec,
+            $fromSpec,
+            $whereSpec
+        );
+
+        // @TODO test this
+        if ($rid = $this->_apiClient->query($countQuery)) {
+            $count = (int)$rid['results']['bindings'];
+            return $count;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Recursively gets owl:imported model IRIs starting with $modelUri as root.
+     *
+     * @param string $modelUri
+     */
+    public function getImportsClosure($modelUri)
+    {
+    	// @TODO test this
+        $queryCache = Erfurt_App::getInstance()->getQueryCache();
+
+        if (!array_key_exists($modelUri, $this->_importedModels)) {
+            $models = array();
+            $result = array(
+                // mock first result
+                array('o' => $modelUri)
+            );
+            do {
+                $from    = '';
+                $filter   = array();
+                foreach ($result as $row) {
+                    $from    .= ' FROM <' . $row['o'] . '>' . PHP_EOL;
+                    $filter[] = 'sameTerm(?model, <' . $row['o'] . '>)';
+
+                    // ensure no model is added twice
+                    if (!array_key_exists($row['o'], $models)) {
+                        $models[$row['o']] = $row['o'];
+                    }
+                }
+                $query = '
+                    SELECT ?o' .
+                    $from . '
+                    WHERE {
+                        ?model <' . EF_OWL_NS . 'imports> ?o.
+                        FILTER (' . implode(' || ', $filter) . ')
+                    }';
+                
+                $result = $queryCache->load($query, Erfurt_Store::RESULTFORMAT_PLAIN);
+                if ($result == Erfurt_Cache_Frontend_QueryCache::ERFURT_CACHE_NO_HIT) {
+                    $startTime = microtime(true);
+                    $result = $this->sparqlQuery($query);
+                    $duration = microtime(true) - $startTime;
+                    $queryCache->save($query, Erfurt_Store::RESULTFORMAT_PLAIN, $result, $duration);
+                }
+            } while ($result);
+
+            // unset root node
+            unset($models[$modelUri]);
+
+            // cache result
+            $this->_importedModels[$modelUri] = array_keys($models);
+        }
+        return $this->_importedModels[$modelUri];
+    } 
+
+    /**
+     * For testing purposes
+     *
+     * @param $httpClientAdapter
+     */
+    public function setHttpClientAdapter($httpClientAdapter)
```

Comment:
We should also remove that test, since the stardog adapter does not handle import? see above comment